subprojects/snapshots/snapshots.gradle.kts

```diff
@@ -62,3 +64,16 @@ gradlebuildJava {
     moduleType = ModuleType.CORE
 }
 
+afterEvaluate {
+    // This is a workaround for the validate plugins task trying to inspect classes which
+    // have changed but are NOT tasks
+    tasks.withType<ValidatePlugins>().configureEach {
+        val main by project.java.sourceSets
+        classes.setFrom(main.output.classesDirs.asFileTree.filter { !it.isInternal(main) })
+    }
+}
+
+fun File.isInternal(sourceSet: SourceSet) = isInternal(sourceSet.output.classesDirs.files)
+
+fun File.isInternal(roots: Set<File>): Boolean = name == "internal" ||
+    !roots.contains(parentFile) && parentFile.isInternal(roots)
```

Comment:
Can we add some parenthesis to make the precedence obvious? Also, in other places we keep boolean operators at the start of the line, not the end of the line. Can we observe that here, too?

Reply:
The method went away.